src/controllers/catalog-controllers/federated-catalog-controller.ts

```diff
@@ -0,0 +1,29 @@
+import { EdcConnectorClientContext } from "../../context";
+import { expandArray, Catalog, QuerySpec } from "../../entities";
+import { Inner } from "../../inner";
+
+export class FederatedCatalogController {
+  #inner: Inner;
+  #context?: EdcConnectorClientContext;
+
+  constructor(inner: Inner, context?: EdcConnectorClientContext) {
+    this.#inner = inner;
+    this.#context = context;
+  }
```

Comment:
`FederatedCatalogController` should extend `EdcController` so we get rid of some boilerplate.